### PR TITLE
WIP: Submitting match by hitting the Enter key

### PR DESCRIPTION
**Pull request by billoosijok**

### CODE REVIEW

File: app/src/components/Forms/MatchContribution/MatchContribution.js [hunk 1]
```diff
@@ -44,19 +44,25 @@ const MatchContribution = ({
...
+  function submitOnEnter(e) {
+    if (e.code === 'Enter') {
+      submitModal();
+      // Removing the Enter event listener after submission
+      document.removeEventListener('keydown', submitOnEnter);
+    }
+  }
 
-  // document.addEventListener('keydown', submitOnEnter);
+  document.addEventListener('keydown', submitOnEnter);
```
Comment (danmelton) [inline app/src/components/Forms/MatchContribution/MatchContribution.js]: let's put this on componentDidMount and remove it on componentUnmount

**Comment by billoosijok** [comment]
@danmelton done 👍.

@kdv24 Can you check over the component please? I converted it to a class-based component, I wanna make sure I didn't screw up any original logic 🙂

**Comment by kdv24** [comment]
I’m out of town till tonight, but can take a look when I get home or tomorrow morning.

**Comment by kdv24** [comment]
@billoosijok thanks for working on this!!  It looks to me like it's all working except for in relation to the `In Compliance` and `Out of Compliance` statuses.  I don't see that it's updating those anymore. 

Although, I wonder if you rebase to `develop` if it will.  I may have fixed that after you last pulled?

**Comment by billoosijok** [comment]
@kdv24 Do you mean that you weren't able to select In Compliance or Out of Compliance? Cuz it's working for me

**Comment by kdv24** [comment]
@billoosijok I think it was that it didn't change the actual status when I looked in the redux tools.  I can check again though, because I'm not sure.

**Comment by kdv24** [comment]
@billoosijok it's the `out of compliance` one.  It's not updating the status in the UI or redux.  I checked `in compliance` and it looks like it's behaving correctly.  Let me know if you find out differently.  I did rebase it to develop before I tested too.

**Comment by billoosijok** [comment]
@kdv24  Does it work for you in the develop branch?

**Comment by kdv24** [comment]
Let me check. Also, I just had thought on it... @billoosijok

**Comment by billoosijok** [comment]
Yeah I tried running it on develop, but it looks like it doesn't work there either

**Comment by kdv24** [comment]
on this branch, it's `processed` but still says `draft` and `compliant: false`:
image
these next two are on develop:
image


image

**Comment by kdv24** [comment]
The only part that I'm seeing that doesn't work is if a record is processed, it returns `[object, Object]`, which I think we just need to get information to know if people should be able to change a record once it's been processed.  If not, we need an error message that's a bit kinder.  Otherwise, we need to update the ability to move out of `processed` status.

**Pull request closed without merging**